xdsl/dialects/arm/ops.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from xdsl.ir import Operation
+from xdsl.irdl import IRDLOperation, irdl_op_definition, result_def
+
+from .register import IntRegisterType
+
+R1InvT = TypeVar("R1InvT", bound=IntRegisterType)
+
+
+class ARMOp(Operation, ABC):
+    """
+    Base class for operations that can be a part of ARM assembly printing.
+    """
+
+    @abstractmethod
+    def assembly_line(self) -> str | None:
+        raise NotImplementedError()
```

Comment:
I don't think there's any need to give an implementation for an abstract method? (I could be very wrong on this, perhaps @superlopuh can correct me)

Reply:
My understanding is that this is good practice. In theory, the annotation should raise an error if not overriden, but I'm not sure if that ever breaks (and what happens if the `@abstractmethod` gets lost somehow.